### DISCUSSION TITLE
Adding a new language?

### DISCUSSION THREAD

**Discussion by qaiwiz** [rhasspy/piper#154] [Ideas]
Is there any good instruction how can I add other lanaguages like Farsi/Persian to piper?

**Comment by Kali7070** [comment]
Would be very interested in that too

**Comment by Francewhoa** [comment]
This documentation might be useful to add other languages at https://github.com/rhasspy/piper/blob/master/TRAINING.md

Video at https://www.youtube.com/watch?v=b_we_jma220

**Reply by qaiwiz** [reply]
nice tutorial. I understand that the higher, the better quality, but is there any good sample txt so that it gives enough data variation for better training. Very much like 20 questions magic ball, that you can ask some kind of least-overlapping questions to get to a target object. Data diversity should be a factor here! Also, imagine you trained something and you created a wav file, which is not good. You can record a wav file to make it better and retrain it. Is there any *active/continuous learning* framework which is based on feedback.

**Comment by SadeghKrmi** [comment]
I've added the model for persian/farsi, please check out https://huggingface.co/rhasspy/piper-voices/tree/main/fa

**Reply by qaiwiz** [reply]
Great. It works but it is not fluent and too many hiccups, specially on pauses. Can you give instruction how to make better, so we can contribute as well?

**Reply by SadeghKrmi** [reply]
in general, this is the matter of datasets, the higher quality of audio, result in the better pronunciation, I tried to feed the higher quality of audio above 13dB. But for Persian/Farsi, there is an extra challenge of diacritic (اعراب گذاری), I would really appreciate if someone could improve the espeak-ng part of the system for Persian/Farsi. 

https://github.com/SadeghKrmi/espeak-ng

**Reply by qaiwiz** [reply]
Yes, I remember in NLP generative models we had same issue. Is there any list of things-to-be-done like q1 mentioned for this?

**Reply by SadeghKrmi** [reply]
for some words, I can see the phonemes are not properly created, which result in incorrect spelling. but some words phoneme are meaningful in the context of the sentence, and some words should pronounced connected to each other, such as "علتِ اصلیٍ بارشِ باران". if you such write "علت اصلی بارش باران", tts is not able to fully catch this connection of words.

Do you have any idea how to solve this in espeak-ng? any other solutions to generate phonemes for persian?

If you like to, please join https://discord.gg/JerZTvVK for build the community and discuss further in discord.

**Reply by SadeghKrmi** [reply]
you can check the phoneme generated by espeak-ng using https://tts.datacula.com/ and then head to the tab "⌨️ Check a word"
